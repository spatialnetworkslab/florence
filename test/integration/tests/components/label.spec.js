/// <reference types="Cypress" />

context('Assertions', () => {
  beforeEach(() => {
    cy.visit('/label-mark')
  })

  describe('Label Mark Tests', () => {
    it('[render] labels rendered', () => {
      cy.get('.label').should('have.length', 4)
    })

    it('[render] labels has correct text', () => {
      cy.get('.label').last().should('contain', 'Testing3')
    })
  })
})
