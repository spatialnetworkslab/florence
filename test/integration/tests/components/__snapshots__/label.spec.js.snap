exports[`Assertions > Label Mark Tests > [render] labels snapshot correctly #0`] = `
<svg width="500" height="500">
  <defs>
    <mask id="mask-sc0-padding-bg">
      <rect width="100%" height="100%" fill="white"></rect>
      <rect x="50" y="50" width="400" height="400" fill="black"></rect>
    </mask>
    <clipPath id="clip-sc0-data">
      <rect x="50" y="50" width="400" height="400"></rect>
    </clipPath>
  </defs>
  <g class="section">
    <text
      clip-path="url(#clip-sc0-data)"
      class="label"
      x="210"
      y="310"
      fill="yellow"
      stroke="none"
      stroke-width="0"
      opacity="1"
      transform="rotate(45, 210, 310)"
      font-family="Georgia"
      font-size="20px"
      font-weight="bold"
      text-anchor="middle"
      dominant-baseline="middle"
    >
      Testing1
    </text>
    <g class="label-layer" clip-path="url(#clip-sc0-data)">
      <text
        class="label"
        x="150"
        y="250"
        fill="black"
        stroke="none"
        stroke-width="0"
        opacity="1"
        transform="
          rotate(45, 
          150, 
          250)
        "
        font-family="Georgia"
        font-size="20px"
        font-weight="bold"
        text-anchor="middle"
        dominant-baseline="middle"
      >
        Testing1
      </text>
      <text
        class="label"
        x="110"
        y="150"
        fill="black"
        stroke="none"
        stroke-width="0"
        opacity="1"
        transform="
          rotate(10, 
          110, 
          150)
        "
        font-family="Georgia"
        font-size="20px"
        font-weight="bold"
        text-anchor="middle"
        dominant-baseline="middle"
      >
        Testing2
      </text>
      <text
        class="label"
        x="90"
        y="90"
        fill="black"
        stroke="none"
        stroke-width="0"
        opacity="1"
        transform="
          rotate(90, 
          90, 
          90)
        "
        font-family="Georgia"
        font-size="20px"
        font-weight="bold"
        text-anchor="middle"
        dominant-baseline="middle"
      >
        Testing3
      </text>
    </g>
  </g>
</svg>
`;
